# Home Credit Risk Analysis

## Dataset: previous_application.csv

### Objetivo

Analizar la estructura, calidad y relaciones del dataset para comprender su papel dentro de la arquitectura Lakehouse y del modelo dimensional.

---

### Contexto del negocio

El dataset previous_application contiene el historial de solicitudes de crédito realizadas previamente por los clientes ante Home Credit.

Cada registro representa una solicitud anterior y almacena información relacionada con montos solicitados, decisiones tomadas, condiciones financieras y características de los productos ofrecidos.

Este dataset permite comprender la relación histórica entre los clientes y la entidad financiera, aportando información valiosa para la evaluación del riesgo crediticio.

---

### Rol dentro de la arquitectura

application_train
        │
        ▼
previous_application

# 1. Importación de librerías

Se cargan las librerías necesarias para el análisis exploratorio de datos.

In [1]:
import pandas as pd
import numpy as np

pd.set_option("display.max_columns", None)
pd.set_option("display.max_rows", 200)

# 2. Carga del Dataset

Se carga el dataset para realizar el análisis exploratorio y comprender su estructura, calidad y relevancia dentro del dominio de negocio.

In [2]:
df = pd.read_csv("../../data/raw/previous_application.csv")

print(f"Filas: {df.shape[0]}")
print(f"Columnas: {df.shape[1]}")

Filas: 1670214
Columnas: 37


# 3. Tamaño del Dataset

## Resultados

| Métrica | Valor |
|----------|---------|
| Filas | 1670214 |
| Columnas | 34 |

## Interpretación

El dataset `previous_application.csv` contiene 1,670,214 registros distribuidos en 37 columnas, convirtiéndose en una de las principales fuentes históricas del ecosistema Home Credit.

El volumen de registros indica que cada cliente puede haber realizado múltiples solicitudes de crédito antes de la solicitud actual registrada en `application_train`.

A diferencia de `bureau.csv`, que almacena créditos reportados por entidades financieras externas, este dataset registra interacciones históricas directamente relacionadas con Home Credit.

### Información contenida

El dataset incorpora información relacionada con:

- Solicitudes de crédito previas.
- Decisiones de aprobación o rechazo.
- Montos solicitados.
- Montos aprobados.
- Productos financieros ofrecidos.
- Canales de adquisición.
- Condiciones contractuales.

### Relevancia para el análisis

El historial de solicitudes constituye una fuente clave para comprender el comportamiento previo de los clientes y evaluar patrones asociados al riesgo crediticio.

A partir de este dataset será posible construir indicadores como:

- Número de solicitudes previas.
- Tasa de aprobación histórica.
- Frecuencia de rechazos.
- Tiempo entre solicitudes.
- Evolución de montos solicitados.

### Rol dentro del ecosistema Home Credit

Mientras que `application_train` representa la solicitud actual del cliente, `previous_application` proporciona contexto histórico sobre su relación previa con la entidad financiera.

Esta información permitirá enriquecer significativamente los modelos analíticos y los indicadores de negocio construidos en las capas Silver, Intermediate, Gold y Diamond.

# 4. Vista Preliminar

Se inspeccionan los primeros registros del dataset con el objetivo de comprender su estructura, contenido y nivel de granularidad.

## Observaciones Iniciales

La muestra evidencia que cada registro corresponde a una solicitud de crédito realizada previamente por un cliente ante Home Credit.

El dataset almacena información relacionada con:

- Tipo de crédito solicitado.
- Montos solicitados y aprobados.
- Estado de la solicitud.
- Condiciones financieras.
- Canal de adquisición.
- Tipo de cliente.
- Información temporal del proceso de aprobación.

---

## Variables Observadas

Entre los atributos más relevantes identificados se encuentran:

| Variable | Descripción |
|-----------|-------------|
| SK_ID_PREV | Identificador de la solicitud previa |
| SK_ID_CURR | Identificador del cliente |
| NAME_CONTRACT_TYPE | Tipo de crédito solicitado |
| AMT_APPLICATION | Monto solicitado |
| AMT_CREDIT | Monto aprobado |
| NAME_CONTRACT_STATUS | Estado de la solicitud |
| CHANNEL_TYPE | Canal de adquisición |
| CNT_PAYMENT | Número de cuotas |
| PRODUCT_COMBINATION | Producto financiero asociado |

---

## Hallazgos Iniciales

La muestra permite identificar múltiples estados posibles para una solicitud:

- Approved
- Refused

Lo anterior sugiere que el dataset contiene tanto solicitudes exitosas como solicitudes rechazadas.

Asimismo, se observan diferencias entre:

- Monto solicitado (`AMT_APPLICATION`)
- Monto aprobado (`AMT_CREDIT`)

lo que indica que Home Credit puede aprobar montos distintos a los originalmente solicitados por el cliente.

---

## Primera Hipótesis de Granularidad

A partir de los registros observados se plantea la siguiente hipótesis:

1 registro = 1 solicitud previa

Esta hipótesis será validada posteriormente mediante el análisis de claves y cardinalidad.

---

## Relación con application_train

Mientras que `application_train` representa la solicitud actual evaluada para el modelo de riesgo, `previous_application` almacena el historial de solicitudes realizadas anteriormente por el cliente.

Esta información permitirá construir indicadores relacionados con:

- Frecuencia de solicitud.
- Historial de aprobación.
- Historial de rechazo.
- Evolución de montos solicitados.
- Relación histórica con la entidad financiera.

In [ ]:
df.head()

# 5. Estructura General

Se analiza:

- Tipos de datos
- Cantidad de columnas
- Valores nulos
- Distribución general

In [ ]:
df.info()

In [ ]:
df.dtypes.value_counts()

## Resumen Estructural

| Métrica | Valor |
|----------|---------|
| Registros | 1,670,214 |
| Columnas | 37 |
| Variables numéricas | 21 |
| Variables categóricas | 16 |
| Tipos de datos distintos | 3 |
| Consumo de memoria | 671.8 MB |

---

## Distribución de Tipos de Datos

| Tipo de dato | Cantidad |
|--------------|----------|
| string | 16 |
| float64 | 15 |
| int64 | 6 |

---

## Interpretación

El dataset presenta una estructura equilibrada entre variables numéricas y categóricas.

Las variables numéricas almacenan principalmente información financiera, temporal y operacional relacionada con las solicitudes de crédito.

Las variables categóricas describen características del proceso de negocio, incluyendo:

- Tipo de contrato.
- Estado de la solicitud.
- Canal de adquisición.
- Tipo de cliente.
- Producto financiero.
- Motivos de rechazo.

Esta combinación proporciona una visión integral tanto del comportamiento financiero como de las decisiones operativas asociadas a cada solicitud.

---

## Componentes Principales

### Variables Financieras

Incluyen información relacionada con:

- Monto solicitado.
- Monto aprobado.
- Valor de los bienes financiados.
- Cuotas.
- Tasas de interés.
- Pagos iniciales.

Estas variables serán fundamentales para construir indicadores financieros históricos.

---

### Variables Temporales

Permiten analizar:

- Fecha relativa de la decisión.
- Inicio del proceso de aprobación.
- Fechas asociadas al ciclo de vida del crédito.

Estas variables serán útiles para construir métricas de comportamiento temporal.

---

### Variables Categóricas

Describen aspectos operativos y comerciales del proceso de crédito.

Su análisis permitirá comprender patrones relacionados con:

- Aprobaciones.
- Rechazos.
- Productos.
- Segmentación de clientes.
- Canales de adquisición.

---

## Calidad Estructural

Se observa que varias columnas financieras presentan cantidades importantes de valores faltantes, especialmente aquellas relacionadas con:

- Tasas de interés.
- Pagos iniciales.
- Fechas de terminación.

Este comportamiento será evaluado con mayor detalle en la sección de calidad de datos.

---

## Consideraciones para Ingeniería de Datos

Con más de 1.6 millones de registros y 37 columnas, previous_application representa una de las principales fuentes históricas del ecosistema Home Credit.

Su nivel de detalle permitirá construir atributos históricos de gran valor para las capas Silver, Intermediate y Gold, especialmente aquellos relacionados con el comportamiento previo de los clientes frente a la entidad financiera.

# 6. Identificación de Claves

Se identifican:

- Clave primaria
- Claves foráneas
- Relaciones con otros datasets

In [6]:
len(df)

1670214

In [7]:
df["SK_ID_PREV"].nunique()

1670214

In [8]:
df["SK_ID_PREV"].duplicated().sum()

np.int64(0)

In [9]:
df["SK_ID_CURR"].nunique()

338857

## Conclusiones sobre las Claves

### Clave Primaria

SK_ID_PREV

La columna SK_ID_PREV identifica de manera única cada solicitud histórica registrada en el dataset.

Las validaciones realizadas muestran que:

- Existen 1,670,214 registros.
- Existen 1,670,214 valores únicos de SK_ID_PREV.
- No se identificaron duplicados.

Por lo tanto, SK_ID_PREV constituye una clave primaria válida.

### Clave Foránea

SK_ID_CURR

Esta columna representa el identificador único del cliente y permite relacionar las solicitudes históricas con la información principal almacenada en application_train.

### Cardinalidad

application_train (1)

↓

previous_application (N)

Un cliente puede realizar múltiples solicitudes de crédito a lo largo del tiempo.

### Granularidad

1 registro = 1 solicitud previa

Cada fila representa una solicitud histórica realizada por un cliente ante Home Credit.

### Interpretación de Negocio

El dataset almacena el historial de solicitudes realizadas por los clientes antes de la solicitud actual.

Esta información permite analizar:

- Frecuencia de solicitud.
- Historial de aprobación.
- Historial de rechazo.
- Comportamiento comercial.
- Relación histórica con Home Credit.

Los resultados obtenidos confirman que previous_application constituye una tabla transaccional histórica orientada al seguimiento de solicitudes de crédito.

# 7. Calidad de Datos

Se analiza la completitud de la información y posibles problemas de calidad.

In [10]:
null_percentage = (
    df.isnull()
      .mean()
      .mul(100)
      .sort_values(ascending=False)
)

null_percentage.head(20)

RATE_INTEREST_PRIVILEGED     99.643698
RATE_INTEREST_PRIMARY        99.643698
AMT_DOWN_PAYMENT             53.636480
RATE_DOWN_PAYMENT            53.636480
NAME_TYPE_SUITE              49.119754
DAYS_TERMINATION             40.298129
DAYS_FIRST_DRAWING           40.298129
DAYS_FIRST_DUE               40.298129
DAYS_LAST_DUE_1ST_VERSION    40.298129
DAYS_LAST_DUE                40.298129
NFLAG_INSURED_ON_APPROVAL    40.298129
AMT_GOODS_PRICE              23.081773
AMT_ANNUITY                  22.286665
CNT_PAYMENT                  22.286366
PRODUCT_COMBINATION           0.020716
AMT_CREDIT                    0.000060
SK_ID_PREV                    0.000000
AMT_APPLICATION               0.000000
NAME_CONTRACT_TYPE            0.000000
SK_ID_CURR                    0.000000
dtype: float64

In [11]:
high_nulls = null_percentage[null_percentage > 50]

medium_nulls = null_percentage[
    (null_percentage > 20)
    & (null_percentage <= 50)
]

low_nulls = null_percentage[
    (null_percentage > 0)
    & (null_percentage <= 20)
]

print("Columnas > 50%:", len(high_nulls))
print("Columnas entre 20%-50%:", len(medium_nulls))
print("Columnas < 20%:", len(low_nulls))

Columnas > 50%: 4
Columnas entre 20%-50%: 10
Columnas < 20%: 2


## Interpretación

El dataset presenta una cantidad considerable de valores faltantes, especialmente en variables relacionadas con tasas de interés, pagos iniciales y fechas asociadas al ciclo de vida del crédito.

Sin embargo, el patrón observado sugiere que gran parte de estos valores faltantes responde a reglas de negocio y no necesariamente a problemas de calidad de datos.

---

### Tasas de Interés

Las variables:

- RATE_INTEREST_PRIMARY
- RATE_INTEREST_PRIVILEGED

presentan más del 99% de valores faltantes.

Esto indica que estas tasas únicamente aplican a un subconjunto muy específico de productos financieros y no constituyen atributos universales para todas las solicitudes.

---

### Información de Pago Inicial

Las variables:

- AMT_DOWN_PAYMENT
- RATE_DOWN_PAYMENT

presentan aproximadamente un 54% de valores faltantes.

Esto puede indicar que una proporción significativa de productos financieros no requiere pago inicial o que dicha información no aplica para determinados tipos de contrato.

---

### Fechas del Ciclo de Vida del Crédito

Las variables:

- DAYS_FIRST_DRAWING
- DAYS_FIRST_DUE
- DAYS_LAST_DUE_1ST_VERSION
- DAYS_LAST_DUE
- DAYS_TERMINATION

comparten exactamente el mismo porcentaje de valores faltantes (40.30%).

Este comportamiento sugiere una dependencia funcional entre ellas y probablemente está relacionado con solicitudes que nunca llegaron a formalizarse o contratos que no alcanzaron determinadas etapas operativas.

---

### Variables Críticas

Las variables clave para la identificación y análisis de las solicitudes presentan niveles prácticamente nulos de información faltante:

- SK_ID_PREV
- SK_ID_CURR
- AMT_APPLICATION
- NAME_CONTRACT_TYPE

Esto garantiza la integridad de los procesos de análisis y modelado posteriores.

---

## Implicaciones para la Arquitectura Lakehouse

### Bronze

Los datos deben preservarse sin modificaciones para mantener la trazabilidad del origen.

### Silver

Será necesario implementar reglas específicas para:

- Identificar nulos por no aplicabilidad.
- Diferenciar nulos operativos de nulos reales.
- Estandarizar variables financieras incompletas.

### Gold

Las agregaciones históricas deberán considerar adecuadamente las columnas con alta proporción de valores faltantes para evitar sesgos en indicadores financieros.

# 8. Detección de Registros Duplicados

In [12]:
df.duplicated().sum()

np.int64(0)

## Hallazgos

### Resultados

| Métrica | Valor |
|----------|---------|
| Registros totales | 1,670,214 |
| Registros duplicados | 0 |
| Porcentaje de duplicados | 0% |

---

### Interpretación

No se identificaron registros completamente duplicados dentro del dataset.

Este resultado confirma la consistencia estructural de la información almacenada y valida la unicidad de las solicitudes históricas representadas por la clave primaria `SK_ID_PREV`.

---

### Conclusión

La ausencia de duplicados garantiza que cada registro corresponde a una solicitud histórica única realizada por un cliente.

Esto favorece la construcción de indicadores históricos y evita sesgos asociados a la duplicación de eventos dentro de los procesos analíticos.

---

### Implicaciones para la Arquitectura Lakehouse

#### Bronze

Los registros pueden almacenarse sin requerir procesos de deduplicación.

#### Silver

No será necesario implementar reglas de eliminación de duplicados para este dataset.

Las validaciones podrán concentrarse en:

- Calidad de datos.
- Consistencia de montos financieros.
- Estados de aprobación y rechazo.
- Integridad referencial con clientes.

#### Gold y Diamond

La ausencia de duplicados garantiza la correcta construcción de métricas históricas relacionadas con:

- Número de solicitudes.
- Tasas de aprobación.
- Tasas de rechazo.
- Comportamiento histórico del cliente.

# 9. Variables de Negocio Relevantes

Se analizan las variables con mayor relevancia para comprender el historial de solicitudes realizadas por los clientes ante Home Credit.

Estas variables permiten evaluar aspectos relacionados con:

- Comportamiento de solicitud.
- Decisiones de aprobación.
- Condiciones financieras.
- Relación histórica con la entidad.
- Riesgo potencial asociado al cliente.

In [ ]:
important_cols = [
    "SK_ID_PREV",
    "SK_ID_CURR",
    "NAME_CONTRACT_TYPE",
    "NAME_CONTRACT_STATUS",
    "AMT_APPLICATION",
    "AMT_CREDIT",
    "CNT_PAYMENT"
]

df[important_cols].head()

In [14]:
(
    df["NAME_CONTRACT_STATUS"]
      .value_counts(normalize=True)
      .mul(100)
      .round(2)
)

NAME_CONTRACT_STATUS
Approved        62.07
Canceled        18.94
Refused         17.40
Unused offer     1.58
Name: proportion, dtype: float64

In [15]:
(
    df["NAME_CONTRACT_TYPE"]
      .value_counts(normalize=True)
      .mul(100)
      .round(2)
)

NAME_CONTRACT_TYPE
Cash loans         44.76
Consumer loans     43.66
Revolving loans    11.57
XNA                 0.02
Name: proportion, dtype: float64

## Variables Identificadas

| Variable | Descripción |
|-----------|-------------|
| SK_ID_PREV | Identificador único de la solicitud previa |
| SK_ID_CURR | Identificador del cliente |
| NAME_CONTRACT_TYPE | Tipo de producto financiero solicitado |
| NAME_CONTRACT_STATUS | Estado final de la solicitud |
| AMT_APPLICATION | Monto solicitado por el cliente |
| AMT_CREDIT | Monto aprobado por la entidad |
| CNT_PAYMENT | Número de cuotas acordadas |

---

## Relevancia de Negocio

### SK_ID_PREV

Permite identificar cada solicitud histórica de manera única.

### SK_ID_CURR

Permite asociar múltiples solicitudes a un mismo cliente y construir métricas históricas agregadas.

### NAME_CONTRACT_TYPE

Describe el tipo de producto financiero solicitado por el cliente.

Esta variable permite identificar preferencias de productos financieros y patrones de consumo.

### NAME_CONTRACT_STATUS

Representa el resultado final de la solicitud.

Es una de las variables más importantes del dataset debido a que permite analizar:

- Solicitudes aprobadas.
- Solicitudes rechazadas.
- Solicitudes canceladas.
- Solicitudes no utilizadas.

### AMT_APPLICATION

Representa el monto solicitado inicialmente por el cliente.

Permite evaluar necesidades financieras y comportamiento de solicitud.

### AMT_CREDIT

Representa el monto finalmente aprobado.

La comparación entre AMT_APPLICATION y AMT_CREDIT permite medir diferencias entre lo solicitado y lo aprobado por la entidad financiera.

### CNT_PAYMENT

Indica la cantidad de cuotas asociadas a la operación financiera.

Esta variable proporciona información sobre el horizonte temporal de pago de los créditos.

---

## Hallazgos Iniciales

La muestra de registros evidencia que el dataset almacena información completa sobre solicitudes históricas realizadas por los clientes.

Se observan diferentes estados de solicitud, incluyendo aprobaciones y rechazos, lo que confirma que el dataset contiene tanto solicitudes exitosas como solicitudes no aprobadas.

Asimismo, se identifican diferencias entre el monto solicitado y el monto finalmente aprobado, sugiriendo la existencia de procesos internos de evaluación y ajuste del riesgo crediticio.

También se observa variabilidad en el número de cuotas, indicando que los productos financieros ofrecidos pueden presentar diferentes horizontes de pago según las características de cada operación.

---

## Distribución de Estados de Solicitud

| Estado | Participación |
|----------|-------------|
| Approved | 62.07% |
| Canceled | 18.94% |
| Refused | 17.40% |
| Unused offer | 1.58% |

### Interpretación

La mayoría de las solicitudes históricas fueron aprobadas, representando el 62.07% del total.

Este resultado sugiere que una proporción significativa de los clientes logró acceder a productos financieros ofrecidos por Home Credit.

Sin embargo, aproximadamente el 38% de las solicitudes no culminaron exitosamente debido a cancelaciones, rechazos u ofertas que no fueron utilizadas.

### Hallazgos

- Más de 6 de cada 10 solicitudes fueron aprobadas.
- Las cancelaciones presentan una frecuencia ligeramente superior a los rechazos.
- Existe evidencia de clientes que iniciaron procesos de crédito sin completar la contratación final.
- El historial de aprobación constituye una fuente valiosa para la construcción de indicadores de comportamiento crediticio.

---

## Distribución de Tipos de Crédito

| Tipo de Crédito | Participación |
|-----------------|-------------|
| Cash loans | 44.76% |
| Consumer loans | 43.66% |
| Revolving loans | 11.57% |
| XNA | 0.02% |

### Interpretación

La distribución de productos financieros se encuentra relativamente equilibrada entre créditos de efectivo y créditos de consumo.

Los préstamos en efectivo representan el 44.76% de las solicitudes, mientras que los créditos de consumo representan el 43.66%.

Los créditos revolventes constituyen una categoría minoritaria, aunque mantienen una participación relevante dentro del portafolio de productos.

### Hallazgos

- Los préstamos de efectivo y de consumo concentran aproximadamente el 88% de todas las solicitudes.
- Los productos revolventes representan una línea de negocio secundaria.
- La categoría XNA presenta una participación insignificante y deberá revisarse durante las etapas posteriores de calidad de datos.

---

## Conclusión

Las variables analizadas evidencian que `previous_application.csv` constituye una fuente estratégica para comprender el historial de interacción entre los clientes y Home Credit.

Los resultados obtenidos permiten identificar patrones relacionados con:

- Frecuencia de aprobación.
- Frecuencia de rechazo.
- Comportamiento de contratación.
- Preferencias de productos financieros.
- Evolución histórica de las solicitudes.

Estas variables serán fundamentales para la construcción de indicadores agregados en las capas Silver, Intermediate y Gold, especialmente aquellos relacionados con el comportamiento histórico del cliente.

Entre los indicadores potenciales destacan:

- Número de solicitudes previas.
- Cantidad de solicitudes aprobadas.
- Cantidad de solicitudes rechazadas.
- Tasa histórica de aprobación.
- Tasa histórica de rechazo.
- Tasa histórica de cancelación.
- Monto promedio solicitado.
- Monto promedio aprobado.
- Preferencia por tipo de producto financiero.

# 10. Variables Categóricas

In [16]:
categorical_columns = df.select_dtypes(include=["object"])

categorical_columns.columns.tolist()

C:\Users\ferna\AppData\Local\Temp\ipykernel_8128\1644596.py:1: Pandas4Warning: For backward compatibility, 'str' dtypes are included by select_dtypes when 'object' dtype is specified. This behavior is deprecated and will be removed in a future version. Explicitly pass 'str' to `include` to select them, or to `exclude` to remove them and silence this warning.
See https://pandas.pydata.org/docs/user_guide/migration-3-strings.html#string-migration-select-dtypes for details on how to write code that works with pandas 2 and 3.
  categorical_columns = df.select_dtypes(include=["object"])


['NAME_CONTRACT_TYPE',
 'WEEKDAY_APPR_PROCESS_START',
 'FLAG_LAST_APPL_PER_CONTRACT',
 'NAME_CASH_LOAN_PURPOSE',
 'NAME_CONTRACT_STATUS',
 'NAME_PAYMENT_TYPE',
 'CODE_REJECT_REASON',
 'NAME_TYPE_SUITE',
 'NAME_CLIENT_TYPE',
 'NAME_GOODS_CATEGORY',
 'NAME_PORTFOLIO',
 'NAME_PRODUCT_TYPE',
 'CHANNEL_TYPE',
 'NAME_SELLER_INDUSTRY',
 'NAME_YIELD_GROUP',
 'PRODUCT_COMBINATION']

In [17]:
for col in categorical_columns.columns:
    print(f"{col}: {df[col].nunique()}")

NAME_CONTRACT_TYPE: 4
WEEKDAY_APPR_PROCESS_START: 7
FLAG_LAST_APPL_PER_CONTRACT: 2
NAME_CASH_LOAN_PURPOSE: 25
NAME_CONTRACT_STATUS: 4
NAME_PAYMENT_TYPE: 4
CODE_REJECT_REASON: 9
NAME_TYPE_SUITE: 7
NAME_CLIENT_TYPE: 4
NAME_GOODS_CATEGORY: 28
NAME_PORTFOLIO: 5
NAME_PRODUCT_TYPE: 3
CHANNEL_TYPE: 8
NAME_SELLER_INDUSTRY: 11
NAME_YIELD_GROUP: 5
PRODUCT_COMBINATION: 17


## Interpretación

El análisis de variables categóricas permitió identificar 16 atributos descriptivos asociados al proceso de solicitud de crédito.

Estas variables capturan información relacionada con productos financieros, comportamiento comercial, canales de adquisición, estados de solicitud y características de los clientes.

Se observa una combinación de variables con baja, media y alta cardinalidad, lo que proporciona un equilibrio adecuado entre capacidad descriptiva y facilidad de análisis.

### Variables de Baja Cardinalidad

Las variables con menor número de categorías incluyen:

- FLAG_LAST_APPL_PER_CONTRACT (2)
- NAME_PRODUCT_TYPE (3)
- NAME_CONTRACT_TYPE (4)
- NAME_CONTRACT_STATUS (4)
- NAME_PAYMENT_TYPE (4)
- NAME_CLIENT_TYPE (4)

Estas variables resultan especialmente útiles para la construcción de indicadores agregados y segmentaciones de negocio.

---

### Variables de Cardinalidad Media

Entre las variables con una cantidad moderada de categorías se encuentran:

- WEEKDAY_APPR_PROCESS_START (7)
- NAME_TYPE_SUITE (7)
- CHANNEL_TYPE (8)
- CODE_REJECT_REASON (9)
- NAME_SELLER_INDUSTRY (11)

Estas variables permiten analizar patrones operativos y comerciales asociados a las solicitudes.

---

### Variables de Alta Cardinalidad

Las variables con mayor diversidad de categorías corresponden a:

- PRODUCT_COMBINATION (17)
- NAME_CASH_LOAN_PURPOSE (25)
- NAME_GOODS_CATEGORY (28)

Estas variables aportan un mayor nivel de detalle sobre las características específicas de los productos financieros y las necesidades de los clientes.

---

### Hallazgos Relevantes

Las variables categóricas con mayor valor analítico identificadas son:

- NAME_CONTRACT_STATUS
- NAME_CONTRACT_TYPE
- CHANNEL_TYPE
- CODE_REJECT_REASON
- NAME_CLIENT_TYPE

Estas variables permitirán construir indicadores relacionados con:

- Tasas de aprobación y rechazo.
- Preferencias de productos financieros.
- Desempeño de canales comerciales.
- Motivos de rechazo de solicitudes.
- Comportamiento histórico de los clientes.

---

### Conclusión

El dataset presenta una estructura categórica rica y diversa que complementa adecuadamente la información financiera contenida en las variables numéricas.

La combinación de atributos comerciales, operativos y financieros convierte a previous_application en una fuente estratégica para comprender el historial de interacción de los clientes con Home Credit y para generar características de alto valor en las capas Silver, Intermediate, Gold y Diamond.

# 11. Relaciones del dataset

## Interpretación

El dataset `previous_application.csv` almacena el historial de solicitudes de crédito realizadas por los clientes antes de la solicitud actual registrada en `application_train`.

Su principal función es proporcionar contexto histórico sobre la relación existente entre cada cliente y Home Credit.

---

### Relación Principal

La relación se establece mediante la columna:

SK_ID_CURR

Esta clave permite conectar las solicitudes históricas con la información principal del cliente.

---

### Cardinalidad

application_train (1)

↓

previous_application (N)

Un cliente puede haber realizado múltiples solicitudes de crédito a lo largo del tiempo.

Cada solicitud histórica pertenece únicamente a un cliente.

---

### Jerarquía de Información

Cliente

↓

Solicitud Actual

↓

Solicitudes Históricas

Esta estructura permite analizar el comportamiento previo del cliente antes de la solicitud evaluada en el modelo de riesgo.

---

### Granularidad

| Dataset | Granularidad |
|----------|-------------|
| application_train | 1 registro = 1 cliente / solicitud actual |
| previous_application | 1 registro = 1 solicitud histórica |

La granularidad observada confirma que previous_application funciona como una tabla transaccional histórica.

---

### Integración con el Modelo Dimensional

El dataset constituye una de las principales fuentes para la construcción de métricas históricas asociadas al cliente.

Entre los indicadores potenciales se encuentran:

- Número total de solicitudes previas.
- Cantidad de solicitudes aprobadas.
- Cantidad de solicitudes rechazadas.
- Cantidad de solicitudes canceladas.
- Tasa histórica de aprobación.
- Tasa histórica de rechazo.
- Tiempo transcurrido desde la última solicitud.
- Monto promedio solicitado.
- Monto promedio aprobado.

---

### Relevancia para la Arquitectura Lakehouse

Durante las etapas Silver e Intermediate, este dataset será utilizado para consolidar información histórica a nivel de cliente.

Posteriormente, las métricas derivadas enriquecerán las capas Gold y Diamond mediante atributos que describen el comportamiento previo de los solicitantes.

---

### Conclusión

previous_application representa la principal fuente de historial interno de solicitudes dentro del ecosistema Home Credit.

Su relación con application_train permite complementar la evaluación actual del cliente mediante información histórica sobre aprobaciones, rechazos, comportamiento comercial y experiencia previa con la entidad financiera.

# 12. Conclusiones Técnicas

## Resumen Ejecutivo

El dataset `previous_application.csv` almacena el historial de solicitudes de crédito realizadas por los clientes ante Home Credit antes de la solicitud actual registrada en `application_train`.

Su principal objetivo es proporcionar contexto histórico sobre la relación comercial existente entre los clientes y la entidad financiera.

El análisis realizado permitió identificar patrones asociados a aprobaciones, rechazos, productos financieros, comportamiento comercial y condiciones crediticias históricas.

---

## Métricas Generales

| Métrica | Valor |
|----------|---------|
| Registros | 1,670,214 |
| Columnas | 37 |
| Clave Primaria | SK_ID_PREV |
| Clave Foránea | SK_ID_CURR |
| Duplicados | 0 |
| Variables Numéricas | 21 |
| Variables Categóricas | 16 |

---

## Calidad de Datos

El dataset presenta una estructura consistente y una adecuada integridad de las columnas críticas para el negocio.

Las validaciones realizadas evidenciaron:

- Ausencia de registros duplicados.
- Clave primaria válida.
- Clave foránea correctamente identificada.
- Variables críticas sin valores faltantes significativos.

Sin embargo, se identificaron niveles elevados de valores nulos en atributos relacionados con:

- Tasas de interés.
- Pagos iniciales.
- Fechas asociadas al ciclo de vida del crédito.

Estos valores faltantes parecen responder principalmente a reglas de negocio y no a problemas de calidad estructural.

---

## Granularidad

Cada registro representa:

**1 solicitud histórica de crédito**

Esta granularidad convierte a previous_application en una tabla transaccional histórica orientada al análisis de comportamiento previo del cliente.

---

## Relaciones Identificadas

### Relación con application_train

application_train (1)

↓

previous_application (N)

Un cliente puede presentar múltiples solicitudes históricas registradas dentro del sistema.

---

### Jerarquía de Información

application_train

↓

previous_application

La relación permite enriquecer la información actual del cliente mediante el análisis de su comportamiento histórico.

---

## Variables de Negocio Relevantes

Las variables de mayor valor analítico identificadas fueron:

- NAME_CONTRACT_STATUS
- NAME_CONTRACT_TYPE
- AMT_APPLICATION
- AMT_CREDIT
- CNT_PAYMENT
- CHANNEL_TYPE
- NAME_CLIENT_TYPE

Estas variables permiten caracterizar el comportamiento histórico de los clientes y construir indicadores asociados al riesgo crediticio.

---

## Hallazgos Principales

### Estados de Solicitud

| Estado | Participación |
|----------|-------------|
| Approved | 62.07% |
| Canceled | 18.94% |
| Refused | 17.40% |
| Unused Offer | 1.58% |

Los resultados muestran que aproximadamente seis de cada diez solicitudes históricas fueron aprobadas.

Asimismo, se observa una proporción relevante de solicitudes canceladas y rechazadas, lo que aporta información valiosa para el análisis de comportamiento crediticio.

---

### Tipos de Crédito

| Tipo | Participación |
|---------|-------------|
| Cash Loans | 44.76% |
| Consumer Loans | 43.66% |
| Revolving Loans | 11.57% |
| XNA | 0.02% |

Los créditos de efectivo y los créditos al consumo concentran la gran mayoría de las solicitudes registradas.

---

## Valor Analítico

El dataset constituye una de las principales fuentes para la construcción de indicadores históricos relacionados con:

- Historial de aprobación.
- Historial de rechazo.
- Frecuencia de solicitud.
- Comportamiento comercial.
- Evolución de montos solicitados.
- Preferencias de productos financieros.

Estos atributos representan una fuente importante de señales para modelos de riesgo y procesos analíticos avanzados.

---

## Rol dentro del Modelo Dimensional

El dataset es candidato natural para la construcción de una tabla de hechos asociada a solicitudes históricas.

### FACT_PREVIOUS_APPLICATION

La información contenida permitirá generar métricas agregadas por cliente tales como:

- Total de solicitudes previas.
- Total de solicitudes aprobadas.
- Total de solicitudes rechazadas.
- Tasa histórica de aprobación.
- Tasa histórica de rechazo.
- Monto promedio solicitado.
- Monto promedio aprobado.
- Tiempo desde la última solicitud.

---

## Rol dentro de la Arquitectura Lakehouse

previous_application.csv

↓

Bronze (ingesta cruda)

↓

Silver (validación y normalización)

↓

Intermediate (agregaciones históricas)

↓

Gold (indicadores de comportamiento)

↓

Diamond (convergencia corporativa)

---

## Recomendaciones para la Siguiente Fase

1. Integrar previous_application con application_train mediante SK_ID_CURR.
2. Construir indicadores históricos de aprobación y rechazo.
3. Analizar la evolución temporal de solicitudes mediante DAYS_DECISION.
4. Crear agregaciones por cliente para la capa Intermediate.
5. Incorporar métricas históricas en las tablas Gold y Diamond.

---

## Conclusión Final

previous_application constituye la principal fuente de historial interno de solicitudes dentro del ecosistema Home Credit.

Su estructura, volumen y riqueza de variables permiten reconstruir el comportamiento histórico de los clientes y generar indicadores de alto valor para procesos analíticos, modelos predictivos y soluciones de inteligencia de negocio.

La integración de este dataset con application_train será fundamental para enriquecer la evaluación del riesgo crediticio mediante información histórica sobre la interacción previa de los clientes con la entidad financiera.